### PR TITLE
Fortnox sync upsert

### PR DESCRIPTION
**Pull request by rippo**
We can now upsert Fortnox customers to webCRM that have changes since a given date

### CODE REVIEW

**Comment by rippo** [comment]
Hey Jan, did you forget to "approve comments" again? Or not had chance to go through this?

File: FortnoxConnector/FortnoxClient.cs [hunk 1]
```diff
@@ -81,5 +75,16 @@ private FilteredCustomersResponse GetActiveCustomers(int page)
 
             return response.Data;
         }
+
+        private RestRequest GetBaseReQuest(int page)
```
Comment (janaagaard75) [inline FortnoxConnector/FortnoxClient.cs]: I think `GetBaseRequest` is a little too broad, and would prefer `CreateFilteredCustomerRequest` or something similar.

File: FortnoxConnector/FortnoxClient.cs [hunk 2]
```diff
@@ -46,6 +46,37 @@ public Customer GetCustomer(string customerNumber)
...
+
+                numberOfPages = response.MetaInformation.TotalPages;
+                currentPageNumber++;
+            } while (currentPageNumber <= numberOfPages);
+        }
+
+        private FilteredCustomersResponse GetRecentlyUpsertedCustomers(DateTime dateSinceUpserted, int page)
+        {
+            var request = GetBaseReQuest(page);
+            request.AddQueryParameter("lastmodified", $"{dateSinceUpserted:yyyy-MM-dd hh:mm}");
```
Comment (janaagaard75) [inline FortnoxConnector/FortnoxClient.cs]: The format string should be `yyyy-MM-dd HH:mm`, with upper case Hs.

File: FortnoxConnector/FortnoxClient.cs [hunk 3]
```diff
@@ -46,6 +46,37 @@ public Customer GetCustomer(string customerNumber)
...
+                {
+                    yield return customer;
+                }
+
+                numberOfPages = response.MetaInformation.TotalPages;
+                currentPageNumber++;
+            } while (currentPageNumber <= numberOfPages);
+        }
+
+        private FilteredCustomersResponse GetRecentlyUpsertedCustomers(DateTime dateSinceUpserted, int page)
```
Comment (janaagaard75) [inline FortnoxConnector/FortnoxClient.cs]: Do we need to handle time zones here, or does that API always use UTC time? If we need to handle time zones, I think we should use DateTimeOffset. If the API expect a UTC time, we could consider naming the parameter something like `dateSinceUpsertedUtc` or `upsertedCutoffDateTimeUtc`.

Comment (rippo) [reply FortnoxConnector/FortnoxClient.cs]: Not sure this is relevant as we will be responsible in passing in a date. It all boils down to how we store the date and retrieve it as the TZ will always be the same for the stored date we hold. Does that make sense?

Comment (rippo) [reply FortnoxConnector/FortnoxClient.cs]: Fortnox makes no mention of what TZ api is in.

https://developer.fortnox.se/documentation/general/encoding/#Date-&%C2%A0Time

Comment (rippo) [reply FortnoxConnector/FortnoxClient.cs]: Also no mention of it on search https://developer.fortnox.se/documentation/general/parameters/

Comment (rippo) [reply FortnoxConnector/FortnoxClient.cs]: But happy to make change or not either way.

Comment (janaagaard75) [reply FortnoxConnector/FortnoxClient.cs]: We have to make sure that our code will work once we set the polling interval down to something like five minutes. I am fine with postponing this until we get to actually start monitoring for changes.

**Pull request merged**